### PR TITLE
Rahul order pt62 pt89

### PR DESCRIPTION
**Pull request by RahulSmartcloudPFQA**
```
Feature: Validate order flow

@OrderWithExistingCreditCard @Smoke @Full 
Scenario: Validate order with Existing valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Compare Product Data On Order Step 2 Page
And Compare Product Data On Order Step 3 Page
And Verify thank you page for credit card transaction
And Verify thank you page data on order history page for credit card transaction
And logout

@OrderWithCreditCardValid @Smoke @Full
Scenario: Validate order with valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Valid Credit Card
And Compare Product Data On Order Step 3 Page
And Verify thank you page for credit card transaction
And Verify thank you page data on order history page for credit card transaction
And logout


@OrderWithCreditCardInValid @Smoke @Full
Scenario: Validate order with In valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With InValid Credit Card
And logout

@OrderWithCreditCardBlankValid @Smoke @Full
Scenario: Validate order with Blank valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Blank Credit Card
And logout

@OrderWithConvenienceStoreValid @Smoke @Full
Scenario: Validate order with valid con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Valid Convenience Store
And Compare Product Data On Order Step 3 Page
And Verify thank you page for convenience store transaction
And Verify thank you page data on order history page for convenience store transaction
And logout

@OrderWithConvenienceStoreInValid @Smoke @Full
Scenario: Validate order with invalid con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With InValid Convenience Store
And logout

@OrderWithConvenienceStoreBlank @Smoke @Full
Scenario: Validate order with Blank Con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Blank Con Store
And logout


@OrderPrimeWithValidCreditCard @Smoke @Full
Scenario: Validate prime with valid Credit Card and check on order history,service you are using and membership page

Given Login To Platform Portal
And Open MemberShip Plan Page from Top page
And Apply MemberShip Plan from Top page
And Compare Prime Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Compare Prime Data On Order Step 2 Page
And Compare Prime Data On Order Step 3 Page
And Verify thank you page for prime transaction
And Verify thank you page data on order history page for prime transaction
And Verify Service you are using page for prime transaction
And Verify Membership status page for prime transaction


@OrderRefreshPageWithExistingCreditCard  @Full
Scenario: Validate refresh page for order with Existing valid Credit Card

Given Login To Platform Portal
And Open Product List Page
And Refresh page
And Select 1st product from product list page
And Verify Order Flow Step 1 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Verify Order Flow Step 2 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 2 Page
And Verify Order Flow Step 3 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 3 Page
And Verify Order Flow Step 4 Thank You Page Loaded Successfully
And Refresh page
And logout


```

### CODE REVIEW

File: platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java [hunk 1]
```diff
@@ -155,27 +143,76 @@ public void loginToPlatform() throws Exception {
...
+//
+//			passwordTextField.sendKeys(shipmentdata[1]);
+//			System.out.println(passwordTextField.getText());
+//		}
+//		if (CommonFunctions.waitForVisiblity(SubmitButton, waitTime)) {
+//			CommonFunctions.clickUsingJavaExecutor(SubmitButton);
+//		}
+//
+//	}
+
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Authentication/LoginPage.java]: Plz remove commented code

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 2]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
 
 	@And("^Step 1 page proceed to next page$")
 	public void step_1_page_proceed_to_next_page() throws Throwable {
-		orderObj.methodForOrderFlowStep1();
-	}
-
-	@And("^Reglogin between Step 1 page to Step 2 page$")
-	public void reglogin_between_step_1_page_to_step_2_page() throws Throwable {
-		orderObj.methodForReloginForOrderFlow();
+		orderObj.methodForOrderFlowStep1(productNameFromProductListPage, productPriceFromProductListPage);
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 3]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
 	}
 
 	@And("^Step 2 page proceed to next page$")
 	public void step_2_page_proceed_to_next_page() throws Throwable {
-		orderObj.methodForOrderFlowStep2();
+		orderObj.methodForOrderFlowStep2(productNameFromProductListPage, productPriceFromProductListPage);
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 4]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
 	}
 
 	@And("^Step 3 page proceed to next page$")
 	public void step_3_page_proceed_to_next_page() throws Throwable {
-		orderObj.methodForOrderFlowStep3();
+		orderObj.methodForOrderFlowStep3(productNameFromProductListPage, productPriceFromProductListPage);
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 5]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Apply MemberShip Plan from Top page$")
+	public void apply_membership_plan_from_top_page() throws Throwable {
+		orderObj.clickApplyForPrimeMembershipButton();
+	}
+
+	@And("^Confirm Order Flow Step 1 Page Load Successfully$")
+	public void confirm_order_flow_step_1_page_load_successfully() throws Throwable {
+		orderObj.orderFlowStep1PageLoadedSuccessfully();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 6]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Confirm Order Flow Step 2 Page Load Successfully$")
+	public void confirm_order_flow_step_2_page_load_successfully() throws Throwable {
+		orderObj.orderFlowStep2PageLoadedSuccessfully();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 7]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Confirm Order Flow Step 3 Page Load Successfully$")
+	public void confirm_order_flow_step_3_page_load_successfully() throws Throwable {
+		orderObj.orderFlowStep3PageLoadedSuccessfully();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 8]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Confirm Order Flow Step 4 thank you Page Load Successfully$")
+	public void confirm_order_flow_step_4_thank_you_page_load_successfully() throws Throwable {
+		orderObj.orderFlowStep4ThankYouPageLoadedSuccessfully();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 9]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Proceeed With 1st Casec Product$")
+	public void proceeed_with_1st_casec_product() throws Throwable {
+		orderObj.proceedWith1stCasecProduct();
+	}
+
+	@And("^Reglogin between Step 1 page to Step 2 page$")
+	public void reglogin_between_step_1_page_to_step_2_page() throws Throwable {
+		orderObj.methodForReloginForOrderFlow();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 10]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
 	}
 
-	@And("^Step2 Payment With Blank Credit Card$")
+	@And("^Step 2 Payment With Valid Credit Card$")
+	public void step2_payment_with_valid_credit_card() throws Throwable {
+		orderObj.methodForStep2PaymentWithValidCreditCard();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 11]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+
+	@And("^Step 2 Payment With Blank Credit Card$")
 	public void step2_payment_with_blank_credit_card() throws Throwable {
 		orderObj.methodForStep2PaymentWithBlankCreditCard();
 	}
 
-	@And("^Step2 Payment With Valid Convenience Store$")
+	@And("^Step 2 Payment With InValid Credit Card$")
+	public void step_2_payment_with_invalid_credit_card() throws Throwable {
+		orderObj.methodForStep2PaymentWithInValidCreditCard();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Order.java [hunk 12]
```diff
@@ -36,44 +49,104 @@ public void select_1st_kanji_product_from_product_list_page() throws Throwable {
...
+	}
+
+	@And("^Step 2 Payment With Valid Convenience Store$")
 	public void step2_payment_with_valid_convenience_store() throws Throwable {
 		orderObj.methodForStep2PaymentWithConStore();
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Order.java]: Please start method name with a verb
Cant determine what this method does
Please make changes wherever applicable

**Review by csonparote — CHANGES_REQUESTED** [review]
Please make changes

**Comment by RahulSmartcloudPFQA** [comment]
```
Feature: Validate order flow

@OrderWithExistingCreditCard @Smoke @Full 
Scenario: Validate order with Existing valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Compare Product Data On Order Step 2 Page
And Compare Product Data On Order Step 3 Page
And Verify thank you page for credit card transaction
And Verify thank you page data on order history page for credit card transaction
And logout

@OrderWithCreditCardValid @Smoke @Full
Scenario: Validate order with valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Valid Credit Card
And Compare Product Data On Order Step 3 Page
And Verify thank you page for credit card transaction
And Verify thank you page data on order history page for credit card transaction
And logout


@OrderWithCreditCardInValid @Smoke @Full
Scenario: Validate order with In valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With InValid Credit Card
And logout

@OrderWithCreditCardBlankValid @Smoke @Full
Scenario: Validate order with Blank valid Credit Card

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Blank Credit Card
And logout

@OrderWithConvenienceStoreValid @Smoke @Full
Scenario: Validate order with valid con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Valid Convenience Store
And Compare Product Data On Order Step 3 Page
And Verify thank you page for convenience store transaction
And Verify thank you page data on order history page for convenience store transaction
And logout

@OrderWithConvenienceStoreInValid @Smoke @Full
Scenario: Validate order with invalid con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With InValid Convenience Store
And logout

@OrderWithConvenienceStoreBlank @Smoke @Full
Scenario: Validate order with Blank Con store

Given Login To Platform Portal
And Select 1st product from product list page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Step 2 Payment With Blank Con Store
And logout


@OrderPrimeWithValidCreditCard @Smoke @Full
Scenario: Validate prime with valid Credit Card and check on order history,service you are using and membership page

Given Login To Platform Portal
And Open MemberShip Plan Page from Top page
And Apply MemberShip Plan from Top page
And Compare Prime Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Compare Prime Data On Order Step 2 Page
And Compare Prime Data On Order Step 3 Page
And Verify thank you page for prime transaction
And Verify thank you page data on order history page for prime transaction
And Verify Service you are using page for prime transaction
And Verify Membership status page for prime transaction


@OrderRefreshPageWithExistingCreditCard  @Full
Scenario: Validate refresh page for order with Existing valid Credit Card

Given Login To Platform Portal
And Open Product List Page
And Refresh page
And Select 1st product from product list page
And Verify Order Flow Step 1 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 1 Page
And Reglogin between Step 1 page to Step 2 page
And Verify Order Flow Step 2 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 2 Page
And Verify Order Flow Step 3 Page Loaded Successfully
And Refresh page
And Compare Product Data On Order Step 3 Page
And Verify Order Flow Step 4 Thank You Page Loaded Successfully
And Refresh page
And logout



```

**Review by csonparote — APPROVED** [review]
Plz merge with main

**Pull request merged**